sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/CollationBenchmark.scala

```diff
@@ -113,13 +113,13 @@ abstract class CollationBenchmarkBase extends BenchmarkBase {
       warmupTime = 10.seconds,
       output = output)
     collationTypes.foreach { collationType => {
-      val collation = CollationFactory.fetchCollation(collationType)
+      val collationId = CollationFactory.collationNameToId(collationType)
```

Comment:
Can we avoid touching the benchmark in the PR, @stevomitric ?

If we need this, we can proceed separately this `CollationBenchmark` change before your regression fix.

Reply:
The regression was partially due to benchmarks as well. The `CollationFactory.collationNameToId` function should only be called fixed amount of times per query, which shouldn't measurably impact the performance of the query.

So i think this is the right PR for it, since it was the leading contributor to the numbers seen in the results.